content/docs/faq-functions.md

```diff
@@ -295,3 +301,58 @@ class Searchbox extends React.Component {
   }
 }
 ```
+
+#### `requestAnimationFrame` throttling
+
+[`requestAnimationFrame`](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) is a way of queuing a function to be executed in the browser at the optimal time for rendering performance. It is very similar to `setTimeout` except it takes no time argument. A function that is queued with `requestAnimationFrame` will fire in the next frame. The browser will work hard to ensure that there are `60` frames per second (`60 fps`). However, if the browser is unable to it will naturally *limit* the amount of frames in a second. For example, a device might only be able to handle `30 fps` and so you will only get `30` frames in that second. Using `requestAnimationFrame` for throttling is a useful technique in that it prevents you from doing more than `60 updates` in a second. If you are doing `100` updates in a second this creates additional work for the browser that the user will not see anyway.
+
+> Using this technique will only capture the last published value in a frame. You can see an example of how this optimization works on [`MDN`](https://developer.mozilla.org/en-US/docs/Web/Events/scroll)
+
+```jsx
+import rafSchedule from 'raf-schd';
+
+class ScrollListener extends React.Component {
+  constructor(props) {
+    super(props);
+    // keep a record of the id of the next animation frame
+    this.nextFrameId = null;
+    this.handleScroll = this.handleScroll.bind(this);
+
+    // create a new function that will schedule updates
+    this.scheduleUpdate = rafSchedule(this.emitUpdate.bind(this));
+  }
+
+  handleScroll(e) {
+    // When we receive a scroll event, we are going to schedule an update
+    // If we receive many updates within a single frame duration we will
+    // only be publishing the latest value.
+
+    this.nextFrameId = this.scheduleUpdate({ x: e.clientX, y: e.clientY });
+  }
+
+  emitUpdate(point) {
+    this.props.onScroll(point);
+  }
+
+  componentWillUnmount() {
+    // As we are unmounted we can clear the next animation frame
+    cancelAnimationFrame(this.nextFrameId);
+  }
+
+  render() {
+    // Creating a scroll container around a big image
+    return (
+      <div
+        style={{ overflow: 'scroll' }}
+        onScroll={this.handleScroll}
+      >
+        <img src="/my-huge-image.jpg" />
+      </div>
+    );
+  }
+}
+```
```

Comment:
I think it might be better for us to use `raf` in this example instead of `raf-schd`.

`raf-schd` looks nice! But `raf` mirrors the behavior of the standardized `requestAnimationFrame` API, and I think it's a good thing for us to show the more standard approach here to avoid potentially confusing users (eg by mixing `raf-schd` and `cancelAnimationFrame`).

Reply:
`raf` is just a `requestAnimationFrame` polyfill and does not contain the logic for queuing calls in a frame. I included `raf-schd` as it is a similar abstraction to a `throttle` or `debounce` function. I could hand write the `requestAnimationFrame` throttle logic, but that seems strange given that we do not do so for `throttle` or `debounce` in this case

Reply:
It would look something like this which is a bit messy:

https://github.com/alexreardon/raf-schd#without-raf-schd

Reply:
> `raf` is just a `requestAnimationFrame` polyfill

Sure, that's true, but I kind of like it for that reason.

> I could hand write the `requestAnimationFrame` throttle logic, but that seems strange given that we do not do so for `throttle` or `debounce` in this case

That's a good point. I think the thing that nagged me a little bit with `raf-schd` was the mix of _that_ module with the built-in `cancelAnimationFrame`.

Reply:
I'm curious- did you ever considered adding a `.cancel` property to the `ResultFn` so as to not have to mix the native and non-native calls?

```js
result.cancel = () => {
  cancelAnimationFrame(frameId);
};
```

Reply:
I did, but that would modify the original function. My thought was to keep it super pure and not modify the wrapped function. However, for consistency with other libs I would consider adding it

Reply:
I am leaning towards adding it to `raf-schd`. It will then make these examples all read very similar

Reply:
Wouldn't it just be a property attached to the inline `result` function you created? It wouldn't actually modify the callback a user passed in right? (I agree, that would be bad.)

Reply:
The original function is not modified - the issue is the flow type. All good, I have been thinking about it for a while. I think having a `.cancel` brings it in line with other utilities. 

Assuming it did add a `.cancel` is there any outstanding issues?

Reply:
Hm. You said:
> I did, but that would modify the original function.

I was just pointing out that the original function (the `fn` callback) wouldn't be modified? Maybe I just misinterpreted what you said. 😄 

No. I think adding a `.cancel` method would alleviate my concerns.

Reply:
https://github.com/alexreardon/raf-schd/pull/7

I will update this pr with the updated syntax

Reply:
Now released is `2.1.0`